src/storage/v2/disk/storage.cpp

```diff
@@ -0,0 +1,1601 @@
+  // vertex. Some other transaction could have modified the vertex in the
+  // meantime if we didn't have any edges to delete.
+
+  if (!PrepareForWrite(&transaction_, vertex_ptr)) return Error::SERIALIZATION_ERROR;
+
+  MG_ASSERT(!vertex_ptr->deleted, "Invalid database state!");
+
+  CreateAndLinkDelta(&transaction_, vertex_ptr, Delta::RecreateObjectTag());
+  vertex_ptr->deleted = true;
+  vertices_to_delete_.emplace_back(utils::SerializeIdType(vertex_ptr->gid), utils::SerializeVertex(*vertex_ptr));
+
+  return std::make_optional<ReturnType>(
+      VertexAccessor{vertex_ptr, &transaction_, &storage_->indices_, &storage_->constraints_, config_, true},
+      std::move(deleted_edges));
+}
+
+void DiskStorage::DiskAccessor::PrefetchEdges(const auto &prefetch_edge_filter) {
+  rocksdb::ReadOptions read_opts;
+  auto strTs = utils::StringTimestamp(transaction_.start_timestamp);
+  rocksdb::Slice ts(strTs);
+  read_opts.timestamp = &ts;
+  auto *disk_storage = static_cast<DiskStorage *>(storage_);
+  auto it = std::unique_ptr<rocksdb::Iterator>(
+      disk_transaction_->GetIterator(read_opts, disk_storage->kvstore_->edge_chandle));
+  for (it->SeekToFirst(); it->Valid(); it->Next()) {
+    const rocksdb::Slice &key = it->key();
+    const auto edge_parts = utils::Split(key.ToStringView(), "|");
+    if (prefetch_edge_filter(edge_parts)) {
+      DeserializeEdge(key, it->value());
+    }
+  }
+}
+
+void DiskStorage::DiskAccessor::PrefetchInEdges(const VertexAccessor &vertex_acc) {
+  PrefetchEdges([&vertex_acc](const std::vector<std::string> &disk_edge_parts) -> bool {
+    auto disk_vertex_in_edge_gid = disk_edge_parts[1];
+    auto edge_gid = disk_edge_parts[4];
+    auto in_edges_res = vertex_acc.InEdges(storage::View::NEW);
+    if (in_edges_res.HasValue()) {
+      for (const auto &edge_acc : in_edges_res.GetValue()) {
+        if (utils::SerializeIdType(edge_acc.Gid()) == edge_gid) {
+          // We already inserted this edge into the vertex's in_edges list.
+          return false;
+        }
+      }
+    }
+    return disk_vertex_in_edge_gid == utils::SerializeIdType(vertex_acc.Gid());
+  });
+}
+
+void DiskStorage::DiskAccessor::PrefetchOutEdges(const VertexAccessor &vertex_acc) {
+  PrefetchEdges([&vertex_acc](const std::vector<std::string> &disk_edge_parts) -> bool {
+    auto disk_vertex_out_edge_gid = disk_edge_parts[0];
+    auto edge_gid = disk_edge_parts[4];
+    auto out_edges_res = vertex_acc.OutEdges(storage::View::NEW);
+    if (out_edges_res.HasValue()) {
+      for (const auto &edge_acc : out_edges_res.GetValue()) {
+        if (utils::SerializeIdType(edge_acc.Gid()) == edge_gid) {
+          // We already inserted this edge into the vertex's out_edges list.
+          return false;
+        }
+      }
+    }
+    return disk_vertex_out_edge_gid == utils::SerializeIdType(vertex_acc.Gid());
+  });
+}
+
+Result<EdgeAccessor> DiskStorage::DiskAccessor::CreateEdge(VertexAccessor *from, VertexAccessor *to,
```

Comment:
I think this function should accept `const VertexAccessor *from`. And similar to `const VertexAccessor *to`